tests/storage/linstor/test_linstor_sr.py

```diff
@@ -0,0 +1,96 @@
+# Requirements:
+# - one XCP-ng host >= 8.2 with an additional unused disk for the SR
+# - access to XCP-ng RPM repository from the host
+
+class TestLinstorSRCreateDestroy:
+    vm = None
+
+    def test_create_sr_without_linstor(self, hosts, lvm_disks):
+        master = hosts[0]
+
+        # This test must be the first in the series in this module
+        assert not master.binary_exists('linstor'), \
+            "linstor must not be installed on the host at the beginning of the tests"
+        try:
+            # though it is expected to fail, result assigned to TestLinstorSR.sr for teardown in case it succeeds
+            sr = master.sr_create('linstor', 'LINSTOR-SR', {
+                'hosts': ','.join([host.hostname for host in hosts]),
+                'group-name': GROUP_NAME,
+                'redundancy': len(hosts),
+                'provisioning': 'thick'
+            }, shared=True)
+            try:
+                sr.destroy()
+            except Exception:
+                pass
+            assert False, "SR creation should not have succeeded!"
+        except CalledProcessError as e:
+            print("SR creation failed, as expected: {}".format(e))
+
+    def test_create_and_destroy_sr(self, linstor_sr, hosts_with_linstor, vm_ref):
+        # Create and destroy tested in the same test to leave the host as unchanged as possible
+        master = hosts_with_linstor[0]
+        # import a VM in order to detect vm import issues here rather than in the vm_on_xfs_fixture used in
+        # the next tests, because errors in fixtures break teardown
+        vm = master.import_vm_url(vm_ref, linstor_sr.uuid)
+        vm.destroy(verify=True)
+
+@pytest.mark.usefixtures("linstor_sr", "vm_on_linstor_sr")
+class TestLinstorSR:
+    def test_start_and_shutdown_VM(self, vm_on_linstor_sr):
+        vm = vm_on_linstor_sr
+        vm.start()
+        vm.wait_for_os_booted()
+        vm.shutdown(verify=True)
+
+    def test_snapshot(self, vm_on_linstor_sr):
+        vm = vm_on_linstor_sr
+        vm.start()
+        vm.wait_for_os_booted()
+        vm.test_snapshot_on_running_linux_vm()
+        vm.shutdown(verify=True)
+
+    # *** tests with reboots (longer tests).
+
+    def test_reboot(self, host, linstor_sr, vm_on_linstor_sr):
+        sr = linstor_sr
+        vm = vm_on_linstor_sr
+        host.reboot(verify=True)
+        wait_for(sr.all_pbds_attached, "Wait for PDB attached")
+        # start the VM as a way to check that the underlying SR is operational
+        vm.start()
+        vm.wait_for_os_booted()
+        vm.shutdown(verify=True)
+
+    def test_linstor_missing(self, linstor_sr, host):
+        packages = ['python-linstor']
+        sr = linstor_sr
+        linstor_installed = True
+        try:
+            host.yum_remove(['python-linstor'])
+            linstor_installed = False
+            try:
+                sr.scan()
+                assert False, "SR scan should have failed"
+            except Exception: # Bad!
```

Comment:
`except subprocess.CalledProcessError:`

Reply:
Oh I missed this fix during my last git push. Thx.